backend/dal/Helpers/Extensions/ProjectExtensions.cs

```diff
@@ -232,5 +231,98 @@ public static void ThrowIfPropertyNotInProjectAgency(this Entity.Project project
                 && property.Agency?.ParentId != project.AgencyId)
                 throw new InvalidOperationException("Properties may not be added to Projects with a different agency.");
         }
+
+        /// <summary>
+        /// Determine the financial year the project is based on.
+        /// Scans properties and gets the most recent evaluation date.
+        /// </summary>
+        /// <param name="project"></param>
+        /// <returns></returns>
+        public static DateTime GetProjectFinancialDate(this Entity.Project project)
```

Comment:
I'm not 100% this is valid, but currently we'll use the most recent financials for the project financial date.